tools/validator-client-scripts/src/main.rs

```diff
@@ -0,0 +1,226 @@
+        Command::Send(args) => send(client, args, &denom).await,
+        Command::DelegateToMixnode(args) => delegate_to_mixnode(client, args, &denom).await,
+        Command::UnDelegateFomMixnode(args) => undelegate_from_mixnode(client, args).await,
+        Command::UpdateProfitPercent(args) => update_profit_percent(client, args).await,
+        Command::VestingUpdateProfitPercent(args) => {
+            vesting_update_profit_percent(client, args).await
+        }
+        Command::VestingBondGateway(args) => vesting_bond_gateway(client, args, &denom).await,
+        Command::VestingBondMixnode(args) => vesting_bond_mixnode(client, args, &denom).await,
+        Command::VestingClaimDelegatorRewards(args) => {
+            vesting_claim_delegator_reward(client, args).await
+        }
+        Command::VestingClaimOperatorRewards(_args) => vesting_claim_operator_reward(client).await,
+        Command::VestingCompoundDelegatorRewards(args) => {
+            vesting_compound_delegator_reward(client, args).await
+        }
+        Command::VestingCompoundOperatorRewards(args) => {
+            vesting_compound_operator_reward(client, args).await
+        }
+        Command::VestingCreateSchedule(args) => vesting_create_schedule(client, args, &denom).await,
+        Command::VestingDelegateToMixnode(args) => {
+            vesting_delegate_to_mixnode(client, args, &denom).await
+        }
+        Command::VestingUnbondGateway(_args) => vesting_unbond_gateway(client).await,
+        Command::VestingUnbondMixnode(_args) => vesting_unbond_mixnode(client).await,
+        Command::VestingUndelegateFromMixnode(args) => {
+            vesting_undelegate_from_mixnode(client, args).await
+        }
+        _ => unreachable!(),
+    }
+}
+
+fn setup_logging() {
+    let mut log_builder = pretty_env_logger::formatted_timed_builder();
+    if let Ok(s) = ::std::env::var("RUST_LOG") {
+        log_builder.parse_filters(&s);
+    } else {
+        // default to 'Info'
+        log_builder.filter(None, log::LevelFilter::Info);
+    }
+
+    log_builder
+        .filter_module("hyper", log::LevelFilter::Warn)
+        .filter_module("tokio_reactor", log::LevelFilter::Warn)
+        .filter_module("reqwest", log::LevelFilter::Warn)
+        .filter_module("mio", log::LevelFilter::Warn)
+        .filter_module("want", log::LevelFilter::Warn)
+        .filter_module("sled", log::LevelFilter::Warn)
+        .filter_module("tungstenite", log::LevelFilter::Warn)
+        .filter_module("tokio_tungstenite", log::LevelFilter::Warn)
+        .init();
+}
+
+fn create_client(args: &Args) -> Client {
+    let network_details = NymNetworkDetails::new_from_env();
+    let client_config = nymd::Config::try_from_nym_network_details(&network_details)
+        .expect("failed to construct valid validator client config with the provided network");
+    NymdClient::connect_with_mnemonic(
+        client_config,
+        args.nymd_url
```

Comment:
`network_details` from above already contains a `nymd_url` from the .env files. I would say, if the url is provided as cli option, use it, but if it's not, don't panic on it with the message `nymd url was not provided` and instead default to the value from .env file.